emulate.c

```diff
@@ -1672,6 +1676,58 @@ static inline bool op_unimp(struct riscv_t *rv, uint32_t insn UNUSED)
     return false;
 }
 
+#ifdef ENABLE_GDBSTUB
+void rv_debug(struct riscv_t *rv)
+{
+    if (!gdbstub_init(&rv->gdbstub, &rv_ops,
+                      (arch_info_t){
+                          .reg_num = 33,
+                          .reg_byte = 4,
+                          .target_desc = TARGET_RV32,
+                      },
+                      SOCKET_ADDR)) {
+        return;
+    }
+
+    if (!gdbstub_run(&rv->gdbstub, (void *) rv)) {
+        return;
+    }
+    gdbstub_close(&rv->gdbstub);
+}
+
+bool rv_get_breakpoint(struct riscv_t *rv, riscv_word_t *addr)
+{
+    assert(rv && addr);
+    if (!rv->breakpoint_specified)
+        return false;
+
+    *addr = rv->breakpoint_addr;
+    return true;
+}
+
+bool rv_set_breakpoint(struct riscv_t *rv, riscv_word_t addr)
```

Comment:
What if we would set multiple breakpoints in GDB? Is it doable?

Reply:
Similarly, is removing breakpoints in GDB functioned?

Reply:
Currently, we will have problems if we want to use more than one breakpoint due to the poor design :(.  Since I thought it could be fine if we just know the restriction in prior, I was planning to improve this in the following patch. But if you think that's a bad idea, I can also spend some time to make this better on this PR.